tools/shared/src/main/scala/org/scalajs/core/tools/jsdep/FlatJSDependency.scala

```diff
@@ -34,4 +34,16 @@ final class FlatJSDependency(
   @deprecated("Use relPath instead.", "0.6.1")
   val resourceName: String = relPath
 
+  override def toString(): String = {
+    val b = new StringBuilder
+    b ++= s"FlatJSDependency(origin=$origin, relPath=$relPath"
+    if (dependencies.nonEmpty)
+      b ++= s", dependencies=$dependencies"
+    if (commonJSName.nonEmpty)
+      b ++= s", commonJSName=$commonJSName"
+    if (relPathMinified.nonEmpty)
+      b ++= s", relPathMinified=$relPathMinified"
```

Comment:
I was following the style of `toString()` methods in other (existing) classes of the [jsdep](https://github.com/scala-js/scala-js/tree/master/tools/shared/src/main/scala/org/scalajs/core/tools/jsdep) package.

Should I change them all?

I'm not convinced by the performance argument, here. These methods are essentially never called, except via a manual interaction with the sbt CLI.

> there is no loss in expressivity or style in this particular case

Could you show an example of the style you would expect?